### ISSUE TITLE
Allow training to be resumed with a different learning rate (or other overridden hyperparameters)

### ISSUE THREAD

**Issue by rubvber** [PyTorchLightning/pytorch-lightning#12118]
### Discussed in https://github.com/PyTorchLightning/pytorch-lightning/discussions/6494

<div type='discussions-op-text'>

<sup>Originally posted by **MightyChaos** March 12, 2021</sup>
I want to resume training from a checkpoint, but I want to use a different learning rate, How to achieve that? I don't  really care about the training states and don't mind start a fresh training as long as the weights are proprely restored.

Right now I'm using ``resume_from_checkpoint=ckpt_file`` when creating the trainer, this automatically would give the old learning rate. 

I also tried remove ``resume_from_checkpoint=ckpt_file``, and do 
```
net_learner.load_from_checkpoint(cfg.ckpt_path, cfg=cfg)
trainer.fit(net_learner, train_data_loader, val_data_loader)
```
but it seems the weights are erased, and the trainer starts from random weights.

Any help will be most appreciated, thanks so much!</div>

## Feature
Provide the ability to resume training a model with a different learning rate (scheduler). Currently, it seems it is only possible within the Lightning framework to resume training from a complete snapshot of a previous state, including not just the model weights and other parameters, but also the optimizer state and any hyperparameters that are set at initialization. 

In parallel, it is possible to load a model from a saved checkpoint, and override any of its hyperparameters for testing or deployment purposes. 

However, a combination of the two does not seem to be possible (but I'd love to be corrected on this). As indicated by the post quoted above, if a model is loaded using `.load_from_checkpoint()`, and then passed to `trainer.fit()`, training does not seem to resume with the saved model weights but with a new initialization. I'm not sure if this is a bug or just the lack of a feature.

### Motivation
It is quite common (in my experience), while experimenting with different learning rate (schedules) for a model, that you want to halt training and resume it with a new learning rate. E.g. maybe the LR scheduler decreased the LR but you want to see what happens if you increase it again, or perhaps continue with a cyclic learning rate. Or maybe you'd like to reduce the learning rate as the loss has plateaud and you didn't account for this in your LR schedule(r). 

More broadly, the model may include hyperparameters that you may want to tweak during a training run, for whatever reason. Maybe you think the model could continue to improve by changing a setting, without having to start from scratch. Again, this is currently possible when using `.load_from_checkpoint()`, but doesn't seem to be an option when resuming training within the Lightning ecosystem. 

Currently, the only workaround for this issue seems to be to load the model using `.load_from_checkpoint()` (overriding any hyperparameters as desired) and then resume training in a bespoke training script (with an optimizer whose learning rate you can control). But again, I'd love to hear if there's a better way within Lightning - if not then I'd like to propose it as a new feature.

**Comment by awaelchli** [comment]
> Anyway, sorry to waste your time and thanks again for taking the time to respond!

No, not at all, don't worry about it. Happy to help.
